tests/compile/test_fusion_attn.py

```diff
@@ -218,24 +98,34 @@ def build_attn_metadata(self, batch_size: int, use_hnd: bool) \
         # Create dummy KV cache for FlashInfer TRTLLM
         #   - NHD: [num_blocks, block_size, num_kv_heads, head_size]
         #   - HND: [num_blocks, num_kv_heads, block_size, head_size]
-        kv_cache = torch.zeros(num_blocks,
-                               2,
-                               self.num_kv_heads,
-                               self.block_size,
-                               self.head_size,
-                               dtype=self.kv_cache_dtype,
-                               device=self.device)
-        if current_platform.is_rocm():
+        if kv_first:
```

Comment:
We only test one of these options per-backend, can we just check the backend here (`self.attn.backend`) to decide which one we're doing, instead of passing the params?